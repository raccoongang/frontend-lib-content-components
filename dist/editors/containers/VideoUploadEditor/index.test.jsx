import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import VideoUploadEditor, { VideoUploader } from '.';
import { formatMessage } from '../../../testUtils';

const defaultEditorProps = {
  onClose: jest.fn().mockName('props.onClose'),
  intl: { formatMessage },
  uploadVideo: jest.fn(),
};

const defaultUploaderProps = {
  onUpload: jest.fn(),
  errorMessage: null,
  intl: { formatMessage },
};

describe('VideoUploader', () => {
  describe('snapshots', () => {
    test('renders as expected with default behavior', () => {
      expect(shallow(<VideoUploader {...defaultUploaderProps} />)).toMatchSnapshot();
    });
    test('renders as expected with error message', () => {
      const defaultUploaderPropsWithError = { ...defaultUploaderProps, errorMessages: 'Some Error' };
      expect(shallow(<VideoUploader {...defaultUploaderPropsWithError} />)).toMatchSnapshot();
    });
  });
});

describe('VideoUploaderEdirtor', () => {
  describe('snapshots', () => {
    test('renders as expected with default behavior', () => {
      expect(shallow(<VideoUploadEditor {...defaultEditorProps} />)).toMatchSnapshot();
    });
  });
});
