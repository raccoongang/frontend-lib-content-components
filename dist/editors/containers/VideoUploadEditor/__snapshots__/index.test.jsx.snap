// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoUploader snapshots renders as expected with default behavior 1`] = `
<div>
  <div
    className="d-flex flex-column justify-content-center align-items-center p-4 w-100 min-vh-100"
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    tabIndex={0}
  >
    <div
      className="d-flex flex-column justify-content-center align-items-center gap-2 text-center min-vh-100 w-100
         dropzone-middle "
    >
      <div
        className="d-flex justify-content-center align-items-center bg-light rounded-circle file-upload"
      >
        <Icon
          className="text-muted"
        />
      </div>
      <div
        className="d-flex align-items-center justify-content-center gap-1 flex-wrap flex-column pt-5"
      >
        <span
          style={
            Object {
              "fontSize": "20px",
            }
          }
        >
          <FormattedMessage
            defaultMessage="Drag and drop video here or click to upload"
            description="Display message for Drag and Drop zone"
            id="VideoUploadEditor.dropVideoFileHere"
          />
        </span>
        <span
          style={
            Object {
              "fontSize": "12px",
            }
          }
        >
          <FormattedMessage
            defaultMessage="Upload MP4 or MOV files (5 GB max)"
            description="Info message for supported formats"
            id="VideoUploadEditor.uploadInfo"
          />
        </span>
      </div>
      <div
        className="d-flex align-items-center mt-3"
      >
        <span
          className="mx-2 text-dark"
        >
          OR
        </span>
      </div>
    </div>
    <input
      accept=""
      data-testid="fileInput"
      multiple={false}
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
  </div>
  <div
    className="d-flex video-id-container"
  >
    <div
      className="d-flex video-id-prompt"
    >
      <input
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Paste your video ID or URL"
        type="text"
        value=""
      />
      <button
        className="border-start-0"
        data-testid="inputSaveButton"
        onClick={[Function]}
        type="button"
      >
        <Icon
          className="rounded-circle text-dark"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`VideoUploader snapshots renders as expected with error message 1`] = `
<div>
  <div
    className="d-flex flex-column justify-content-center align-items-center p-4 w-100 min-vh-100"
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    tabIndex={0}
  >
    <div
      className="d-flex flex-column justify-content-center align-items-center gap-2 text-center min-vh-100 w-100
         dropzone-middle "
    >
      <div
        className="d-flex justify-content-center align-items-center bg-light rounded-circle file-upload"
      >
        <Icon
          className="text-muted"
        />
      </div>
      <div
        className="d-flex align-items-center justify-content-center gap-1 flex-wrap flex-column pt-5"
      >
        <span
          style={
            Object {
              "fontSize": "20px",
            }
          }
        >
          <FormattedMessage
            defaultMessage="Drag and drop video here or click to upload"
            description="Display message for Drag and Drop zone"
            id="VideoUploadEditor.dropVideoFileHere"
          />
        </span>
        <span
          style={
            Object {
              "fontSize": "12px",
            }
          }
        >
          <FormattedMessage
            defaultMessage="Upload MP4 or MOV files (5 GB max)"
            description="Info message for supported formats"
            id="VideoUploadEditor.uploadInfo"
          />
        </span>
      </div>
      <div
        className="d-flex align-items-center mt-3"
      >
        <span
          className="mx-2 text-dark"
        >
          OR
        </span>
      </div>
    </div>
    <input
      accept=""
      data-testid="fileInput"
      multiple={false}
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
  </div>
  <div
    className="d-flex video-id-container"
  >
    <div
      className="d-flex video-id-prompt"
    >
      <input
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Paste your video ID or URL"
        type="text"
        value=""
      />
      <button
        className="border-start-0"
        data-testid="inputSaveButton"
        onClick={[Function]}
        type="button"
      >
        <Icon
          className="rounded-circle text-dark"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`VideoUploaderEdirtor snapshots renders as expected with default behavior 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
